src/StreamJsonRpc.Shared/Reflection/TargetMethod.cs

```diff
@@ -28,12 +32,43 @@ internal TargetMethod(JsonRpcMessage request, object target, JsonSerializer json
             this.target = target;
 
             var targetMethods = new Dictionary<MethodSignature, object[]>();
+            bool errorFound = false;
             for (TypeInfo t = target.GetType().GetTypeInfo(); t != null; t = t.BaseType?.GetTypeInfo())
             {
-                bool matchFound = false;
-                foreach (MethodInfo method in t.GetDeclaredMethods(request.Method))
+                var matchFound = false;
+                var methodName = request.Method;
+                var methodAttribute = request.Method;
+
+                if (methodToAttributeMap != null)
+                {
+                    foreach (string methodKey in methodToAttributeMap.Keys)
+                    {
+                        if (methodToAttributeMap[methodKey].Any(a => string.Equals(a, request.Method, StringComparison.OrdinalIgnoreCase)))
```

Comment:
It feels like we're using the dictionary backwards, IMO. All access to it is driven by incoming messages which will contain the *value* of the dictionary entry instead of the key. But imagine if we swapped key/value in this dictionary: then we could do a constant time lookup to see if there is a substitution we should apply to the incoming method name.